constraint/pkg/apis/constraints/apis.go

```diff
@@ -45,3 +64,101 @@ func GetEnforcementAction(constraint *unstructured.Unstructured) (string, error)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %w", ErrInvalidConstraint, err)
+	}
+
+	// Flag to indicate if all enforcement points should be enforced
+	enforceAll := false
+	// Initialize a map to hold enforcement actions for each enforcement point
+	actionsForEPs := make(map[string]map[string]bool)
+	// Populate the actionsForEPs map with enforcement points from eps, initializing their action maps
+	for _, enforcementPoint := range eps {
+		if enforcementPoint == AllEnforcementPoints {
+			enforceAll = true // Set enforceAll to true if the special identifier for all enforcement points is found
+		}
+		actionsForEPs[enforcementPoint] = make(map[string]bool) // Initialize the action map for the enforcement point
+	}
+
+	// Iterate over the scoped enforcement actions to populate actions for each enforcement point
+	for _, scopedEA := range scopedEnforcementActions {
+		for _, enforcementPoint := range scopedEA.EnforcementPoints {
+			epName := strings.ToLower(enforcementPoint.Name)
+			ea := strings.ToLower(scopedEA.Action)
+			// If enforceAll is true, or the enforcement point is explicitly listed, initialize its action map
+			if _, ok := actionsForEPs[epName]; !ok && enforceAll {
+				actionsForEPs[epName] = make(map[string]bool)
+			}
+			// Skip adding actions for enforcement points not in the list unless enforceAll is true
+			if _, ok := actionsForEPs[epName]; !ok && epName != AllEnforcementPoints {
+				continue
+			}
+			// If the enforcement point is the special identifier for all, apply the action to all enforcement points
+			switch epName {
+			case AllEnforcementPoints:
+				for ep := range actionsForEPs {
+					actionsForEPs[ep][ea] = true
+				}
+			default:
+				actionsForEPs[epName][ea] = true
+			}
+		}
+	}
+
+	return actionsForEPs, nil
+}
+
+// Helper function to access nested fields as an array.
+func getNestedFieldAsArray(obj map[string]interface{}, fields ...string) ([]interface{}, bool, error) {
+	value, found, err := unstructured.NestedFieldNoCopy(obj, fields...)
+	if err != nil {
+		return nil, false, err
+	}
+	if !found {
+		return nil, false, nil
+	}
+	if arr, ok := value.([]interface{}); ok {
+		return arr, true, nil
+	}
+	return nil, false, nil
+}
+
+// Helper function to convert a value to a []ScopedEnforcementAction.
+func convertToSliceScopedEnforcementAction(value interface{}) ([]ScopedEnforcementAction, error) {
+	var result []ScopedEnforcementAction
+	if arr, ok := value.([]interface{}); ok {
+		for _, v := range arr {
+			if m, ok := v.(map[string]interface{}); ok {
+				scopedEA := &ScopedEnforcementAction{}
+				if err := runtime.DefaultUnstructuredConverter.FromUnstructured(m, scopedEA); err != nil {
+					return nil, err
+				}
+				result = append(result, *scopedEA)
+			} else {
+				return nil, fmt.Errorf("scopedEnforcementActions value must be a []scopedEnforcementAction{action: string, enforcementPoints: []EnforcementPoint{name: string}}")
```

Comment:
You can consider checking these conditions first, then return. this will reduce the number of elses you need in the code.